JASP-Engine/JASP/R/contingencytables.R

```diff
@@ -15,1778 +15,1412 @@
+    
+    res <- try(.crossTabOddsRatioRows(jaspResults, analysis$rows, 
+                                      analyses$group.matrices[[i]], 
+                                      analyses$groups[[i]], analysisContainer, options, ready))
+    .crossTabSetErrorOrFill(res, crossTabLogOdds)
+  }
+}
 
-				ss.dataset <- base::subset(dataset, select=.v(c(rows, columns, counts)))
+.crossTabNominal <- function(jaspResults, dataset, options, analyses, ready) {
+  if (!options$contingencyCoefficient && !options$phiAndCramersV && !options$lambda)
+    return()
+  for (i in 1:length(analyses$analyses)){
+    analysis <- analyses$analyses[[i]]
+    analysisContainer <- jaspResults[[paste0("tables", i)]]
+    if (!is.null(analysisContainer[["crossTabNominal"]])) 
+      next
+    
+    # Create table
+    crossTabNominal <- createJaspTable(title = "Nominal")
+    crossTabNominal$dependOn(c("contingencyCoefficient", "phiAndCramersV"))
+    crossTabNominal$showSpecifiedColumnsOnly <- TRUE
+    crossTabNominal$position <- 4
+    
+    # Add columns to table
+    .crossTabLayersColumns(crossTabNominal, analysis)
+    if (options$contingencyCoefficient){
+      crossTabNominal$addColumnInfo(name = "type[ContCoef]",  title = "", 
+                                    type = "string")
+      crossTabNominal$addColumnInfo(name = "value[ContCoef]", title = "Value", 
+                                    type = "number")
+    }
+    
+    if (options$phiAndCramersV) {
+      crossTabNominal$addColumnInfo(name = "type[PhiCoef]",  title = "", type = "string")
+      crossTabNominal$addColumnInfo(name = "value[PhiCoef]", title = "Value", 
+                                    type = "number")
+      crossTabNominal$addColumnInfo(name = "type[CramerV]",  title = "", type = "string")
+      crossTabNominal$addColumnInfo(name = "value[CramerV]", title = "Value", 
+                                    type = "number")
+    }
+    
+    if (options$lambda) {
+      crossTabNominal$addColumnInfo(name = "type[LambdaR]",  title = "", type = "string")
+      crossTabNominal$addColumnInfo(name = "value[LambdaR]", title = "Value", 
+                                    type = "number")
+      crossTabNominal$addColumnInfo(name = "type[LambdaC]",  title = "", type = "string")
+      crossTabNominal$addColumnInfo(name = "value[LambdaC]", title = "Value", 
+                                    type = "number")
+    }
+    
+    analysisContainer[["crossTabNominal"]] <- crossTabNominal
+
+    res <- try(.crossTabNominalRows(jaspResults, analysis$rows, 
+                                    analyses$group.matrices[[i]], 
+                                    analyses$groups[[i]], analysisContainer, options, ready))
+    .crossTabSetErrorOrFill(res, crossTabNominal)
+  }
+}
 
-			} else {
+.crossTabGamma <- function(jaspResults, dataset, options, analyses, ready) {
+  if (!options$gamma)
+    return()
+  for (i in 1:length(analyses$a
```

Comment:
here us some code duplication.. maybe you could use a function to do:

```r
if (inherits(col.proportions.matrix[1,1], "character"))
  col.proportions <- col.proportions.matrix[1,]
else 
  colproportion <- apply(col.proportions.matrix, 2, base::sum)

col.proportions <- as.list(colproportion)
names(col.proportions) <- paste(names(col.proportions), "[col.proportions]", sep = "")

if (inherits(row.proportions.matrix[1,1], "character")) {
  col.proportions[["total[col.proportions]"]] <- ""
```
which is common to all 4 if statements.